### PR TITLE
Add truncation capability as to not create invalid messages

### PR DESCRIPTION
**Pull request by RoccoMathijn**
Hey, I added a capability to truncate fields according to the specified start and end positions

Lmk what you think

### CODE REVIEW

**Comment by RoccoMathijn** [comment]
Rebasing on https://github.com/atais/Fixed-Length/pull/16 will fix the build. Waiting for a review on that

File: src/test/scala/com/github/atais/fixedlength/simple/EncoderTest.scala [hunk 1]
```diff
@@ -28,8 +28,10 @@ class EncoderTest extends AnyFlatSpec with Matchers {
 
     implicit val employeeCodec: Encoder[Employee] = {
       fixed[String](0, 10) <<:
-        fixed[Option[Int]](10, 13, Alignment.Right) <<:
-        fixed[Boolean](13, 18)
+          fixed[Option[Int]](10, 13, Alignment.Right) <<:
+          fixed[Boolean](13, 18) <<:
+          fixed[String](18, 24, truncate = Truncation.Right) <<:
```
Comment (atais) [inline src/test/scala/com/github/atais/fixedlength/simple/EncoderTest.scala:33]: can you please test all the possibilities:

|Alignment/Truncation  |Left   |Right   |
|---|---|---|
|Left   | :heavy_check_mark:  | :heavy_check_mark:  |
|Right   | :heavy_check_mark:  | :white_check_mark:   |

**Comment by atais** [comment]
LGTM. 

Overall I see why you need it ;-)
But...

On the other hand maybe it would be better to make Encoder returning `Try` or `Either` and be able to react on those errors?
WDYT?

File: src/main/scala/com/github/atais/fixedlength/Truncation.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.github.atais.fixedlength
+
+sealed trait Truncation
```
Comment (atais) [inline src/main/scala/com/github/atais/fixedlength/Truncation.scala:3]: please update readme

File: src/test/scala/com/github/atais/fixedlength/simple/EncoderTest.scala (hunk 1 above)
Comment (RoccoMathijn) [reply src/test/scala/com/github/atais/fixedlength/simple/EncoderTest.scala:33]: ✅

**Comment by RoccoMathijn** [comment]
> LGTM.
> 
> Overall I see why you need it ;-) But...
> 
> On the other hand maybe it would be better to make Encoder returning `Try` or `Either` and be able to react on those errors? WDYT?

I Like that approach too. However the codecs now are a really nice grammar specification of a fixed length line/file. If we throw an encoding error it would mean we would have to move field truncation to a different place where we need all the information on field lengths again.

In my case, where I'm working on a fixed length file with 140+ fields that would get real messy. 

Or did you have something else in mind?

File: src/main/scala/com/github/atais/fixedlength/Truncation.scala (hunk 2 above)
Comment (RoccoMathijn) [reply src/main/scala/com/github/atais/fixedlength/Truncation.scala:3]: ✅

**Comment by atais** [comment]
> Or did you have something else in mind?

That was my idea exactly.

It's hard for me to tell how is this lib actually behaving in long term or what would come useful because I have used it only for a month or two in a project and it was abandoned.

So I am glad to see someone finds it useful and is willing to put some work to make it better! Thanks

**Pull request merged**